debridge_node/src/modules/jobs/services/actions/CheckAssetsEventAction.ts

```diff
@@ -198,14 +198,21 @@ export class CheckAssetsEventAction extends IAction {
     this.logger.log(`Finish Check assets event`);
   }
 
-  private async getTokenInfo(nativeChainId: number, nativeTokenAddress: string) {
-    const tokenChainDetail = this.chainConfigService.get(nativeChainId) as EvmChainConfig;
+  async #getEvmTokenInfo(chainId: number, tokenAddress: string, abi: any = ERC20Abi) {
+    const tokenChainDetail = this.chainConfigService.get(chainId) as EvmChainConfig;
     const tokenWeb3 = await this.web3Service.web3HttpProvider(tokenChainDetail);
-    const nativeTokenInstance = new tokenWeb3.eth.Contract(ERC20Abi as any, nativeTokenAddress);
-    const tokenName = await getTokenName(nativeTokenInstance, nativeTokenAddress, { logger: this.logger });
-    const tokenSymbol = await nativeTokenInstance.methods.symbol().call();
+    const nativeTokenInstance = new tokenWeb3.eth.Contract(abi, tokenAddress);
     const tokenDecimals = await nativeTokenInstance.methods.decimals().call();
-    // nativeAdress = nativeTokenInfo.nativeAddress;
-    return { tokenName, tokenSymbol, tokenDecimals };
+    try {
+      const [tokenName, tokenSymbol] = await Promise.all([nativeTokenInstance.methods.name().call(), nativeTokenInstance.methods.symbol().call()]);
+
+      return { tokenName, tokenSymbol, tokenDecimals };
+    } catch (e) {
+      if (e.message.includes('NUMERIC_FAULT')) {
+        return await this.#getEvmTokenInfo(chainId, tokenAddress, DSToken);
```

Comment:
this method should return string human readable. We need to convert byte32 to string human readable